plugin/advisor/mysql/advisor_naming_index_convention.go

```diff
@@ -0,0 +1,200 @@
+package mysql
+
+import (
+	"fmt"
+	"regexp"
+	"strings"
+
+	"github.com/bytebase/bytebase/api"
+	"github.com/bytebase/bytebase/common"
+	"github.com/bytebase/bytebase/plugin/advisor"
+	"github.com/bytebase/bytebase/plugin/db"
+	"github.com/pingcap/tidb/parser/ast"
+)
+
+var (
+	_ advisor.Advisor = (*NamingIndexConventionAdvisor)(nil)
+)
+
+func init() {
+	advisor.Register(db.MySQL, advisor.MySQLNamingIndexConvention, &NamingIndexConventionAdvisor{})
+	advisor.Register(db.TiDB, advisor.MySQLNamingIndexConvention, &NamingIndexConventionAdvisor{})
+}
+
+// NamingIndexConventionAdvisor is the advisor checking for index naming convention.
+type NamingIndexConventionAdvisor struct {
+}
+
+// Check checks for index naming convention.
+func (check *NamingIndexConventionAdvisor) Check(ctx advisor.Context, statement string) ([]advisor.Advice, error) {
+	root, errAdvice := parseStatement(statement, ctx.Charset, ctx.Collation)
+	if errAdvice != nil {
+		return errAdvice, nil
+	}
+
+	level, err := advisor.NewStatusBySchemaReviewRuleLevel(ctx.Rule.Level)
+	if err != nil {
+		return nil, err
+	}
+
+	format, templateList, err := api.UnmarshalNamingRulePayloadAsTemplate(ctx.Rule.Type, ctx.Rule.Payload)
+	if err != nil {
+		return nil, err
+	}
+	checker := &namingIndexConventionChecker{
+		level:        level,
+		format:       format,
+		templateList: templateList,
+	}
+	for _, stmtNode := range root {
+		(stmtNode).Accept(checker)
+	}
+
+	if len(checker.adviceList) == 0 {
+		checker.adviceList = append(checker.adviceList, advisor.Advice{
+			Status:  advisor.Success,
+			Code:    common.Ok,
+			Title:   "OK",
+			Content: "",
+		})
+	}
+	return checker.adviceList, nil
+}
+
+type namingIndexConventionChecker struct {
+	adviceList   []advisor.Advice
+	level        advisor.Status
+	format       string
+	templateList []string
+}
+
+func (checker *namingIndexConventionChecker) Enter(in ast.Node) (ast.Node, bool) {
+	indexDataList := checker.getMetaDataList(in)
+
+	for _, indexData := range indexDataList {
+		regex, err := getTemplateRegexp(checker.format, checker.templateList, indexData.metaData)
+		if err != nil {
+			checker.adviceList = append(checker.adviceList, advisor.Advice{
+				Status:  checker.level,
+				Code:    common.Internal,
+				Title:   "Internal error for index naming convention rule",
+				Content: fmt.Sprintf("%q meet internal error %q", in.Text(), err.Error()),
+			})
+			continue
+		}
+		if !regex.MatchString(indexData.index) {
+			checker.adviceList = append(checker.adviceList, advisor.Advice{
+				Status:  checker.level,
+				Code:    common.NamingIndexConventionMismatch,
+				Title:   "Mismatch index naming convention",
+				Content: fmt.Sprintf("%q mismatches index naming convention, expect %q but found %q", in.Text(), checker.format, indexData.index),
```

Comment:
```suggestion
				Content: fmt.Sprintf("%q mismatches index naming convention, expect %q but found `%s`", in.Text(), checker.format, indexData.index),
```
After using backticks, we can ignore the reserved keyword case, which I think is more canonical for MySQL.
Details https://dev.mysql.com/doc/refman/8.0/en/keywords.html and https://dev.mysql.com/doc/refman/8.0/en/identifiers.html.

Reply:
An additional discussion, `in.Text()` sometimes is too long, such as `CREATE TABLE` statement. How about using `Index in Table mismatches index naming convention, naming format should be %q`?